### PR TITLE
SourceLayout: Refactor comm_incoming mechanism

### PR DESCRIPTION
**Pull request by yadij**

### CODE REVIEW

File: doc/release-notes/release-7.sgml.in [hunk 1]
```diff
@@ -151,6 +151,12 @@ This section gives an account of those changes in three categories:
 	<tag>--without-psapi</tag>
 	<p>Disable auto-detection of Windows PSAPI library.
 
+	<tag>CPPFLAGS=-DINCOMING_FACTOR=</tag>
+	<p>Control the listening sockets responsiveness.
+	   The higher the INCOMING_FACTOR, the slower the algorithm will
+	   respond to load spikes/increases/decreases in demand. A value
+	   between 3 and 8 is recommended. Default is 5.
```
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:158]: Please consider mentioning that this option is specific to poll(2) and select(2) descriptor polling methods?

**Review by rousskov — COMMENTED** [review]
I assume these changes only affect Squids that are using poll(2) or select(2). Please let me know if my assumption in wrong.

FWIW, I do not intend to review this PR details under the above assumption: I see code that should probably be written differently, but it feels wrong to spend resources on optimizing poll(2) and select(2) performance. Do many admins that care about performance at this level actually use those descriptor polling mechanisms instead of their (usually faster) modern alternatives like epoll(2)?

File: src/comm/Incoming.h [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * Caveats:
+ *
+ *   \copydoc INCOMING_FACTOR
+ *
+ *   \copydoc Comm::Incoming::MaxInterval
+ */
+class Incoming
+{
+public:
+#if !defined(INCOMING_FACTOR)
```
Comment (kinkie) [inline src/comm/Incoming.h:42]: Would it be possible to use a const integer type instead, and move the preprocessor-based default value setting outside the class?

**Review by kinkie — COMMENTED** [review]
I like the concept; I think it needs polishing before merging, but it's a nice step forward

File: src/comm/Incoming.h (hunk 2 above)
Comment (yadij) [reply src/comm/Incoming.h:42]: Done what can be done in [e2da9e8](https://github.com/squid-cache/squid/pull/1572/commits/e2da9e8d68417728303809b330d9873db8f7f7af). The `#ifdef` is inside the class to make it clear that the magic value is only relevant in that scope.

**Comment by yadij** [comment]
Clearing for merge under the "10 + days with no objections" criteria.

**Pull request closed without merging**